docker/Dockerfile

```diff
@@ -0,0 +1,67 @@
+################################################################################
+# Build stage 0 `builder`:
+# Install OpenSearch Benchmark from source inside a virtualenv
+################################################################################
+
+FROM python:3.11.2-slim as builder
+
+RUN apt-get -y update && \
+    apt-get install -y curl git gcc && \
+    apt-get -y upgrade && \
+    rm -rf /var/lib/apt/lists/*
+
+WORKDIR /opensearch-benchmark/
+COPY ./setup.py /opensearch-benchmark/
+COPY ./setup.cfg /opensearch-benchmark/
+COPY ./version.txt /opensearch-benchmark/
+COPY ./README.md /opensearch-benchmark/
+COPY ./MANIFEST.in /opensearch-benchmark/
+COPY ./osbenchmark/ /opensearch-benchmark/osbenchmark/
+
+RUN python3 -m venv /opensearch-benchmark/venv
+ENV PATH="/opensearch-benchmark/venv/bin:$PATH"
+
+WORKDIR /opensearch-benchmark
+# Wipe away any lingering caches, copied over from the local machine
+RUN find /opensearch-benchmark -name "__pycache__" -exec rm -rf -- \{\} \; 2>/dev/null || true
+RUN find /opensearch-benchmark -name ".pyc" -exec rm -rf -- \{\} \; 2>/dev/null || true
+RUN python3 -m pip install --upgrade pip setuptools wheel
+RUN python3 -m pip install /opensearch-benchmark
```

Comment:
In the context of the Dockerfile, it's the correct syntax since we are using `WORKDIR` in the following fashion prior to line 26:
```
WORKDIR /opensearch-benchmark
```
We're instructing docker to run `pip3 install` on the working directory.